phpBB/phpbb/auth/provider/db.php

```diff
@@ -153,8 +153,12 @@ public function login($username, $password)
 		}
 
 		$login_error_attempts = 'LOGIN_ERROR_ATTEMPTS';
-		$show_captcha = ($this->config['max_login_attempts'] && $row['user_login_attempts'] >= $this->config['max_login_attempts']) ||
-			($this->config['ip_login_limit_max'] && $attempts >= $this->config['ip_login_limit_max']);
+
+		$user_login_attempts	= $row ? ($this->config['max_login_attempts'] && $row['user_login_attempts'] >= $this->config['max_login_attempts']) : false;
+		$ip_login_attempts		= (bool) ($this->config['ip_login_limit_max'] && $attempts >= $this->config['ip_login_limit_max']);
```

Comment:
I don't think you need the bool cast.

Reply:
Probably not but I do not see anything against it, I can remove it if so you wish though.

Reply:
The `&&` operator produces a `bool`. So the cast is redundant.

Reply:
Okay.